python/ray/dag/compiled_dag_node.py

```diff
@@ -243,6 +235,85 @@ def __str__(self) -> str:
 """
 
 
+@DeveloperAPI
```

Comment:
I think these should not be developer APIs. Prepending the class name with underscore will allow you to not specify an API decorator.